### PR TITLE
Give usefull error when Nodejs exe is corrupt

### PR DESCRIPTION
**Pull request by paulvanbrenk**
Fixes issue #103

Catches the exception and gives a helpful errormessage

### CODE REVIEW

File: Nodejs/Product/PressAnyKey/PressAnyKey.cs [hunk 1]
```diff
@@ -19,22 +20,35 @@ private static int Main(string[] args)
...
+            try
+            {
+                var proc = Process.Start(psi);
+                File.WriteAllText(args[1], proc.Id.ToString());
+                proc.WaitForExit();
+                exitCode = proc.ExitCode;
+            }
+            catch (Win32Exception exc)
+            {
+                Console.WriteLine($"Failed to start process: '{exc.Message}'.");
```
Comment (amcasey) [inline Nodejs/Product/PressAnyKey/PressAnyKey.cs]: Do these strings need to be localized?

File: Nodejs/Product/PressAnyKey/PressAnyKey.cs [hunk 2]
```diff
@@ -19,22 +20,35 @@ private static int Main(string[] args)
...
+                Console.WriteLine("Probable cause is the Nodejs exe is corrupt, please re-install.");
```
Comment (amcasey) [inline Nodejs/Product/PressAnyKey/PressAnyKey.cs]: If we're displaying this to the user, we probably have to be more careful about names.  I think they use "Node.js".

File: Nodejs/Product/PressAnyKey/PressAnyKey.cs [hunk 3]
```diff
@@ -19,22 +20,35 @@ private static int Main(string[] args)
...
+                exitCode = -1;
+            }
 
             if (args[0] == "both" ||
-                (proc.ExitCode == 0 && args[0] == "normal") ||
-                (proc.ExitCode != 0 && args[0] == "abnormal"))
+                (exitCode == 0 && args[0] == "normal") ||
```
Comment (amcasey) [inline Nodejs/Product/PressAnyKey/PressAnyKey.cs]: `args[0] == (exitCode == 0 ? "normal" : "abnormal")`?

File: Nodejs/Product/PressAnyKey/PressAnyKey.cs [hunk 4]
```diff
@@ -19,22 +20,35 @@ private static int Main(string[] args)
...
 
-            var proc = Process.Start(psi);
-            File.WriteAllText(args[1], proc.Id.ToString());
-            proc.WaitForExit();
+            var exitCode = 0;
+            try
+            {
+                var proc = Process.Start(psi);
+                File.WriteAllText(args[1], proc.Id.ToString());
+                proc.WaitForExit();
```
Comment (amcasey) [inline Nodejs/Product/PressAnyKey/PressAnyKey.cs]: This can also throw `Win32Exception`.

File: Nodejs/Product/PressAnyKey/PressAnyKey.cs (hunk 1 above)
Comment (amcasey) [inline Nodejs/Product/PressAnyKey/PressAnyKey.cs]: Apparently, nothing else in this assembly is.

File: Nodejs/Product/PressAnyKey/PressAnyKey.cs (hunk 4 above)
Comment (amcasey) [inline Nodejs/Product/PressAnyKey/PressAnyKey.cs]: Error message to be updated.

File: Nodejs/Product/PressAnyKey/PressAnyKey.cs (hunk 2 above)
Comment (amcasey) [inline Nodejs/Product/PressAnyKey/PressAnyKey.cs]: Error message to be updated.

File: Nodejs/Product/PressAnyKey/PressAnyKey.cs (hunk 4 above)
Comment (paulvanbrenk) [inline Nodejs/Product/PressAnyKey/PressAnyKey.cs]: I think in that case your windows is broken...

File: Nodejs/Product/PressAnyKey/PressAnyKey.cs (hunk 1 above)
Comment (paulvanbrenk) [inline Nodejs/Product/PressAnyKey/PressAnyKey.cs]: as discussed

File: Nodejs/Product/PressAnyKey/PressAnyKey.cs (hunk 2 above)
Comment (paulvanbrenk) [inline Nodejs/Product/PressAnyKey/PressAnyKey.cs]: changed

**Pull request merged**